packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/SVGComponents/VXZoom/VXZoom.js

```diff
@@ -7,22 +7,29 @@ import ZoomEventLayer from '../ZoomEventLayer'
 class VXZoom extends PureComponent {
   constructor (props) {
     super(props)
-    const { setOnZoom } = props
+    const { setOnPan, setOnZoom } = props
 
+    setOnPan(this.handleToolbarPan.bind(this))
     setOnZoom(this.handleToolbarZoom.bind(this))
```

Comment:
Instead of calling these on mount, maybe call them when a chart is selected and actively being controlled, so that only one chart is registered with the toolbar at any one time?

Reply:
The `VXZoom` component is agnostic to what viewer is using it and currently it is a parent component for `ScatterPlot` component when `zooming` is true. It doesn't directly wrap or interact with the `VariableStarViewer`. I think calling these on focus of the parent would couple it to the `VariableStarViewer` with refs, which I don't think I want to do.

Reply:
I was thinking that each scatter plot viewer, when selected, could call `setOnPan` and `setOnZoom` to register itself with the toolbar as the component that is currently being controlled. That's more in line with how those functions were designed to be used but not needed for this PR.

Reply:
Have a look sometime at the scatter plot viewer code. `VXZoom` is only parent to the `ScatterPlot` when `zooming` is true, so in effect, these are only being called when the user selects to control it.